### PR TITLE
add documentatition for Setting Up Client Network Chapter

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds documentation for Setting Up Client Network chapter in the issue #340.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.redoOperation = true;
+```
+
+Its default value is `false` (disabled).
+
+## 4. Setting Connection Timeout
+
+Connection timeout is the timeout value in milliseconds for members to accept client connection requests. The following are the example configurations.
```
Comment (mustafaiman) [inline README.md]: What happens if the server does not respond within connection timeout? Does client quit trying altogether?

File: README.md [hunk 2]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.addresses.push('10.1.1.21', '10.1.1.22:5703');
+```
+
+If the port part is omitted, then 5701, 5702 and 5703 will be tried in random order.
+
+You can provide multiple addresses with ports provided or not, as seen above. The provided list is shuffled and tried in random order. Its default value is `localhost`.
+
+## 2. Setting Smart Routing
+
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: I will explain it in the `Using Node.js Client with Hazelcast IMDG`chapter like Java reference manual.

File: README.md (hunk 1 above)
Comment (furkansenharputlu) [reply README.md]: I referenced Java documentation. See http://docs.hazelcast.org/docs/3.10.4/manual/html-single/index.html#setting-connection-timeout

File: README.md [hunk 3]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.addresses.push('10.1.1.21', '10.1.1.22:5703');
+```
+
+If the port part is omitted, then 5701, 5702 and 5703 will be tried in random order.
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: I am not sure but it is already written so in the CONFIG.md. See https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/CONFIG.md#configuring-address-list

File: README.md [hunk 4]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+
+You can provide multiple addresses with ports provided or not, as seen above. The provided list is shuffled and tried in random order. Its default value is `localhost`.
```
Comment (Serdaro) [inline README.md]: You can specify multiple addresses with or without port information as seen above. The...

File: README.md [hunk 5]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+
+## 2. Setting Smart Routing
+
+Smart routing defines whether the client mode is smart or unisocket. The following are example configurations.
```
Comment (Serdaro) [inline README.md]: Make a new line for the sentence "The following are example configurations."

And before that give a cross-reference to the operation modes section:

"See Node.js Client Operation Modes (hyperlinked) section for the description of smart and unisocket modes."

File: README.md (hunk 1 above)
Comment (Serdaro) [reply README.md]: Furkan, if you know the answer to this, you should either include the link, or the actual content here which is specific to Node.js.

File: README.md [hunk 6]
```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
...
+var clientConfig = new Config.ClientConfig();
+clientConfig.groupConfig.name = 'hazel';
+clientConfig.groupConfig.password = 'cast';
+
+clientConfig.networkConfig.cloudConfig.enabled = true;
+clientConfig.networkConfig.cloudConfig.discoveryToken = 'EXAMPLE_TOKEN';
+```
+
+To be able to connect to the provided IP addresses, you should use secure TLS/SSL connection between the client and members. Therefore, you should set an SSL configuration as described in the previous section.
+
```
Comment (Serdaro) [inline README.md]: Let's take a note to reference to the "hazelcast.cloud" documentation when it is ready.

File: README.md (hunk 1 above)
Comment (furkansenharputlu) [reply README.md]: I will add the answer.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: I gave a cross-reference to the operation modes section as @Serdaro said.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mustafaiman) [reply README.md]: Can you make sure what is written in `CONFIG.md` is correct and fix if neccessary?

[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: Yes, this feature is not supported in Node.js. I will open an issue about it.

[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: @mustafaiman I added this behaviour in a new PR(#368).

File: README.md [hunk 7]
```diff
@@ -145,6 +153,239 @@ readData(dataInput) {
 }
 ```
 
+# Setting Up Client Network
+
+All network related configuration of Hazelcast Node.js Client is performed via the `network` element in the declarative configuration file, or in the object ClientNetworkConfig` when using programmatic configuration. Let’s first give the examples for these two approaches. Then we will look at its sub-elements and attributes.
```
Comment (mustafaiman) [inline README.md]: "ClientNetworkConfig\`" is missing a "`" just before it.

File: README.md [hunk 8]
```diff
@@ -145,6 +153,239 @@ readData(dataInput) {
...
+clientConfig.networkConfig.addresses.push('10.1.1.21', '10.1.1.22:5703');
+```
+
+If the port part is omitted, then 5701, 5702 and 5703 will be tried in random order.
+
+You can specify multiple addresses with or without port information as seen above. The provided list is shuffled and tried in random order. Its default value is `localhost`.
+
+## 2. Setting Smart Routing
+
+Smart routing defines whether the client mode is smart or unisocket. See [Node.js Client Operation Modes section](#nodejs-client-opertaion-modes)
```
Comment (mustafaiman) [inline README.md]: `#nodejs-client-opertaion-modes` - > `#nodejs-client-operation-modes`

File: README.md [hunk 9]
```diff
@@ -145,6 +153,239 @@ readData(dataInput) {
...
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.connectionAttemptPeriod = 5000;
+```
+
+Its default value is `3000` milliseconds.
+
+## 7. Enabling Hazelcast Cloud Discovery
+
+The purpose of Hazelcast Cloud Discovery is to provide clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to "true".
```
Comment (mustafaiman) [inline README.md]: "true" -> \`true\`

**Pull request merged**